DOC: add the doc source file for the `yulesimon`

rlucas7 opened the pull request:
partially closes #9997 , adds a doc source for `yulesimon`.

rlucas7 commented:
The html build works locally but my latex build does not. The latex issues (I see 12 locally) first appear in the `discrete_laplace.rst` file when I do a latex build locally. 

I'm not seeing helpful error logging in the CI build that helps me determine if the issue is the same or something else. 

It seems like this is an issue from a different commit than those on the PR here?

rgommers commented:
> It seems like this is an issue from a different commit than those on the PR here?

I just tried building master and that works fine. I'll try your branch too.

rgommers commented:
Hmm, `make all-pdf` worked fine on master but with your branch it indeed stops in `dlaplace`:
```
LaTeX Warning: Hyper reference `generated/scipy.stats.dlaplace:scipy.stats.dlap
lace' on page 332 undefined on input line 27119.

! Missing $ inserted.
<inserted text> 
                $
l.27126  p\left
               (k;\alpha\right) = \alpha \B\left(k, \alpha+1\right),
```
It may just be that adding `yulesimon` right below `dlaplace` triggered a difference in behavior for a missing `$` in `dlaplace`.

ilayn commented:
I think it is `\B` that is undefined here..

rlucas7 commented:
Ilayn, 

I think you are right. I looked in beta and beta prime docs and looks like the beta function is never demoted directly. Instead always through the use of the gamma functions. I will rewrite in terms of gamma functions directly when I get home this evening.

Thanks! 

Sincerely, 

-Lucas Roberts 

> On Apr 13, 2019, at 12:19 PM, Ralf Gommers <notifications@github.com> wrote:
> 
> It seems like this is an issue from a different commit than those on the PR here?
> 
> I just tried building master and that works fine. I'll try your branch too.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

rlucas7 commented:
@ilayn and @rgommers Looks like switching `\B(a,b)` to the ratio of gamma functions removed the doc build failure.

pvanmulbregt commented:
Many of the tutorials for the stats distributions follow a similar format.  See https://docs.scipy.org/doc/scipy/reference/tutorial/stats/discrete_bernoulli.html for an example.
(The consistency of the discrete tutorials is not as good as it is for the continuous ones.)
Do you think you could modify this tutorial to match that layout?

ilayn commented:
Also you could have just type `B(a, b)` or `\beta(a,b)`. I just wanted to point that `\B` is not a valid macro in LaTeX. But if gamma definition is better nice to have consistency.

rlucas7 commented:
Yeah it looks like the normalizing constant in the binomial cumulative mass function uses the ratio of gamma functions so this should be consistent with what I see in other discrete distributions. 

Sincerely, 

-Lucas Roberts 

> On Apr 13, 2019, at 12:19 PM, Ralf Gommers <notifications@github.com> wrote:
> 
> It seems like this is an issue from a different commit than those on the PR here?
> 
> I just tried building master and that works fine. I'll try your branch too.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

pvanmulbregt commented:
I like the reformatting. 
The build failure appears to be real, as it also failed when I tried to use it locally. Something in the first TeX eqnarray block is triggering an error.

pvanmulbregt commented on doc/source/tutorial/stats/discrete_yulesimon.rst:
```diff
@@ -0,0 +1,41 @@
+random variate generation.
+
+.. math::
+
```
The `math::` block below has `:nowrap:`.  This one doesn't. Perhaps it needs it?

rlucas7 commented:
The build issue on commit ` 4dacf4b` is weird, the log states a missing `\right)` but I they all look balanced to me and the build completed locally in my dev environment just fine. 

I know when I run `python setup.py build_sphinx` it is building the html versions of the files, i'm not sure if the html build uses latex or not, presumably `sphinx` is using latex under the hood but it is difficult for me to ascertain if my assumption is true. The issues in building seem to be solely from latex.

rlucas7 commented:
ci/circleci: pypy3  build failure looks unrelated (numpy issue?). 

Otherwise this fixes the issues, the docs build successfully. I added a `:nowrap:` as suggested by @pvanmulbregt and fixed the unmatched `\left(` by removing a `\left)` and making it a `\right)`. 

Let me know if anything else is needed for the tutorial doc. Would be good to have this in so the distribution is known and used.

ilayn commented:
pypy run on the CI is removed, you can ignore it

rlucas7 commented:
Anything else needed to get this PR merged? 

Sincerely, 

-Lucas Roberts 

> On Apr 26, 2019, at 2:09 AM, Ilhan Polat <notifications@github.com> wrote:
> 
> pypy run on the CI is removed, you can ignore it
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub, or mute the thread.

pvanmulbregt commented:
Thanks @rlucas7. (Nothing else needed, the main line has just opened up again after the freeze related to the 1.3 release.)

rlucas7 commented:
Ah, I didn’t know one can freeze a branch in github. Good to know. Thanks for merging-and the feedback. 

Sincerely, 

-Lucas Roberts 

> On Apr 27, 2019, at 1:25 PM, Paul van Mulbregt <notifications@github.com> wrote:
> 
> Thanks @rlucas7. (Nothing else needed, the main line has just opened up again after the freeze related to the 1.3 release.)
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
